index.js

```diff
@@ -32,21 +43,47 @@ function spawnClangFormat(args, done, stdio) {
               "Consider installing it with your native package manager instead.\n";
     throw new Error(message);
   }
-  var child_process = spawn(nativeBinary, args, {stdio: stdio});
-  child_process.on('close', function(exit) {
+  var clangFormatProcess = spawn(nativeBinary, args, {stdio: stdio});
+  clangFormatProcess.on('close', function(exit) {
     if (exit) done(exit);
   });
-  return child_process;
+  return clangFormatProcess;
 }
 
-module.exports = clangFormat;
-module.exports.spawnClangFormat = spawnClangFormat;
-
-if (require.main === module) {
+function main() {
+  // Find clang-format in node_modules of the project of the .js file, or cwd.
+  var nonDashArgs = process.argv.filter(function(arg, idx) { return idx > 1 && arg[0] != '-' });
+  // Using the last file makes it less likely to collide with clang-format's argument parsing.
+  var lastFileArg = nonDashArgs[nonDashArgs.length - 1];
+  var basedir = lastFileArg ? path.dirname(lastFileArg) :  // relative to the last .js file given.
```

Comment:
opt: while you're doing relative path stuff here, it would also be really nice to print a command that gives relative paths to the files to format. I tried that once but had a bug in some corner case.